### PR TITLE
feat: add age pref in profile form

### PR DESCRIPTION
**Pull request by Mikazian**
## Description

- Add to the profile form, the age pref form

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Area of change

- [x] Frontend
- [x] Backend
- [ ] Other: .....

## General checklist:

- [x] My code follows the style guidelines of this project
- [x] I ran `bun run format`/`bun run lint` before commit
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation where needed
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] I have updated `develop` and merged to my branch before submitting pull request
- [x] My pull request generate no conflicts with `develop` branch
- [x] I requested code review from other team members

## Frontend checklist:

- [ ] I followed guidelines for `HTML`/`JSX`/`Typescript`
- [ ] My `Javascript` generate no new console errors
- [ ] I tested my code cross browsers
- [ ] My slice is pixel perfect for both desktop and mobile according to design
- [ ] I conducted basic QA to assure all features are working
- [ ] I tested responsive for mobile and tablet resolutions

## Backend checklist:

- [ ] I tested admin by manually adding content from zero
- [ ] I created easy to use admin experience which is self-explanatory
- [ ] I added description to admin fields in hard-to-understand areas
- [ ] I followed guidelines for naming `Typescript` variables
- [ ] I conducted basic QA to assure all features are working

### CODE REVIEW

**Review by greegouze — APPROVED** [review]

**Review by greegouze — DISMISSED** [review]

**Review by greegouze — DISMISSED** [review]

File: packages/frontend/web/src/components/forms/FormAge.tsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  register('maxAge', {
+    valueAsNumber: true,
+    validate: (value) => {
+      const result = maxAgeSchema.safeParse(value);
+      if (!result.success) {
+        return result.error.errors[0].message;
+      }
+      return true;
+    },
+  });
```
Comment (YDrogen) [inline packages/frontend/web/src/components/forms/FormAge.tsx:41]: ??? Pourquoi ??

File: packages/frontend/web/src/pages/FormProfile.tsx [hunk 2]
```diff
@@ -73,16 +81,27 @@ export default function FormProfile() {
...
         });
 
-        setIsLoggedIn(true);
-        navigate('/');
+        if (res.ok) {
+          const controller = new AbortController();
+          const signal = controller.signal;
+
+          fetchPreferences({ signal });
+          await fetchUsers({ signal });
```
Comment (YDrogen) [inline packages/frontend/web/src/pages/FormProfile.tsx:97]: Pourquoi y'en a qu'un en `await` ?

**Review by YDrogen — CHANGES_REQUESTED** [review]

File: packages/frontend/web/src/pages/FormProfile.tsx (hunk 2 above)
Comment (Mikazian) [reply packages/frontend/web/src/pages/FormProfile.tsx:97]: Capture d’écran 2024-02-27 à 14 20 42

Comment (Mikazian) [reply packages/frontend/web/src/pages/FormProfile.tsx:97]: j'ai changé le typage de return en void du coup j'ai retiré les await

File: packages/frontend/web/src/components/forms/FormAge.tsx (hunk 1 above)
Comment (Mikazian) [reply packages/frontend/web/src/components/forms/FormAge.tsx:41]: retiré j'ai créé des inputs cachés

**Review by greegouze — APPROVED** [review]

**Review by YDrogen — APPROVED** [review]

**Pull request merged**